tst/data/proxy/UserProfileStoreTest.java

```diff
@@ -1,5 +1,66 @@
 package data.proxy;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import data.structure.UserProfile;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
 public class UserProfileStoreTest {
-    // TODO
+
+    /**
+     * Tests null return for missing UserProfile id.
+     */
+    @Test
+    public void testGetProfileNonExistent() {
+        TestUserProfileStore userProfileStore = new TestUserProfileStore();
+        Map<String, UserProfile> userProfiles = new HashMap<String, UserProfile>();
+        userProfileStore.setUserProfiles(userProfiles);
```

Comment:
I don't think this is really necessary here. If you're trying to test that a null is returned when "get" is called with a non-existent ID, then you can just create a UserProfileStore instance and immediately call the get method. Furthermore, this is the only case where you use the TestUserProfileStore class (and by proxy, the protected variable "userProfiles"), so I don't think that this is worth expanding the visibility of the private userProfiles attribute of UserProfileStore.

I would advise resetting that attribute's visibility to private and removing the TestUserPRofileStore class here for the sake of better encapsulation and simpler code.

P.S. - Even if you did leave it as protected (not saying you should), you don't have to have a separate test class just for that. Even UserProfileStoreTest could access protected variables.